### PR TITLE
[fulen] Fix GDB 7.10.1 and add GBD 8.1.1

### PR DESCRIPTION
**Pull request by victorusu**
There is a missing patch for GDB 7.10.1

### CODE REVIEW

**Comment by victorusu** [comment]
retest this please

File: easybuild/easyconfigs/g/GDB/GDB-8.1.1.eb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+version = '8.1.1'
+
+homepage = 'http://www.gnu.org/software/gdb/gdb.html'
+description = "The GNU Project Debugger"
+
+# http://mirror.switch.ch/ftp/mirror/gnu/gdb/gdb-7.10.1.tar.gz
+source_urls = ['http://mirror.switch.ch/ftp/mirror/gnu/gdb/']
+sources = [SOURCELOWER_TAR_XZ]
+
+toolchain = {'name': 'dummy', 'version': 'dummy'}
```
Comment (gppezzi) [inline easybuild/easyconfigs/g/GDB/GDB-8.1.1.eb]: version should be empty or dependencies are not loaded

Comment (victorusu) [reply easybuild/easyconfigs/g/GDB/GDB-8.1.1.eb]: I made it `dummy`, because I think GDB should not be a dependency ever.
But I can change it to ''.

Comment (gppezzi) [reply easybuild/easyconfigs/g/GDB/GDB-8.1.1.eb]: Let me rephrase that: the tc version should be empty (i'm not referring to the name) or the dependencies (inside this eb file) are not loaded (unless eb has recently changed this behaviour recently, this recipe is using ncurses from the system and not from eb).

Comment (victorusu) [reply easybuild/easyconfigs/g/GDB/GDB-8.1.1.eb]: I think that setting `'version': 'dummy'` affects the use of the `GBD` as a dependency, but not the use of dependencies within `GDB`. Because I can see the `module load ncurses` command inside the generated `GDB` module
```
module show GDB/8.1.1 
-------------------------------------------------------------------
/apps/.../GDB/8.1.1:

module-whatis	 Description: The GNU Project Debugger 
module-whatis	 Homepage: http://www.gnu.org/software/gdb/gdb.html 
conflict	 GDB 
module		 load ncurses/.6.0 
...
```

Comment (gppezzi) [reply easybuild/easyconfigs/g/GDB/GDB-8.1.1.eb]: Did you check also during the load occurs during the build process? 

EB not loading the dependencies when the toolchain version is defined is a known issue and we had to chang many recipes because of this. 

If this behaviour changed, we should confirm (ideally asking @boegel and identifying the PR that touched that) and then inform all the team.

Comment (boegel) [reply easybuild/easyconfigs/g/GDB/GDB-8.1.1.eb]: Using `dummy` as version means that the dependencies won't be loaded during the build, but the will be included in the generated module file, see also https://easybuild.readthedocs.io/en/latest/Writing_easyconfig_files.html#loading-of-modules-for-dependencies-with-a-dummy-toolchain .

Long story short: always use `''` as version when using `dummy` toolchain. The behavior with the `dummy` version is a historical, which we intend to rectify with the `system` toolchain concept which is supposed to replace `dummy`, see https://github.com/easybuilders/easybuild-framework/issues/1181 (it'll happen some day, will put this back on the agenda for EB v4.0 (see https://github.com/easybuilders/easybuild/issues/447)).
